Modules/TRD/src/TrackletsTask.cxx

```diff
@@ -294,92 +385,116 @@ void TrackletsTask::drawHashedOnHistsPerLayer(int iLayer)
+  if (auto param = mCustomParameters.find("markerstyle");
+      param != mCustomParameters.end()) {
     mMarkerStyle = stof(param->second);
-    ILOG(Info, Support) << "configure() : using marketstyle : = " << mMarkerStyle << ENDM;
+    ILOG(Info, Support) << "configure() : using marketstyle : = "
+                        << mMarkerStyle << ENDM;
   } else {
     mMarkerStyle = 3; // a plus sign
-    ILOG(Info, Support) << "configure() : using default dritfregionstart = " << mMarkerStyle << ENDM;
+    ILOG(Info, Support) << "configure() : using default dritfregionstart = "
+                        << mMarkerStyle << ENDM;
   }
-  if (auto param = mCustomParameters.find("markersize"); param != mCustomParameters.end()) {
+  if (auto param = mCustomParameters.find("markersize");
+      param != mCustomParameters.end()) {
     mMarkerSize = stof(param->second);
-    ILOG(Info, Support) << "configure() : using markersize : = " << mMarkerSize << ENDM;
+    ILOG(Info, Support) << "configure() : using markersize : = " << mMarkerSize
+                        << ENDM;
   } else {
     mMarkerSize = 3; // a plus sign
-    ILOG(Info, Support) << "configure() : using default markersize = " << mMarkerSize << ENDM;
+    ILOG(Info, Support) << "configure() : using default markersize = "
+                        << mMarkerSize << ENDM;
   }
 
   retrieveCCDBSettings();
   buildHistograms();
 }
 
-void TrackletsTask::startOfActivity(Activity& activity)
-{
+void TrackletsTask::startOfActivity(Activity &activity) {
   ILOG(Info, Support) << "startOfActivity " << activity.mId << ENDM;
   for (Int_t sm = 0; sm < o2::trd::constants::NSECTOR; ++sm) {
     moHCMCM[sm]->Reset();
   }
 }
 
-void TrackletsTask::startOfCycle()
-{
+void TrackletsTask::startOfCycle() {
   ILOG(Info, Support) << "startOfCycle" << ENDM;
 }
 
-void TrackletsTask::monitorData(o2::framework::ProcessingContext& ctx)
-{
-  for (auto&& input : ctx.inputs()) {
+void TrackletsTask::monitorData(o2::framework::ProcessingContext &ctx) {
+  for (auto &&input : ctx.inputs()) {
     if (input.header != nullptr && input.payload != nullptr) {
 
       auto digits = ctx.inputs().get<gsl::span<o2::trd::Digit>>("digits");
-      auto tracklets = ctx.inputs().get<gsl::span<o2::trd::Tracklet64>>("tracklets");
-      auto triggerrecords = ctx.inputs().get<gsl::span<o2::trd::TriggerRecord>>("triggers");
+      auto tracklets =
+          ctx.inputs().get<gsl::span<o2::trd::Tracklet64>>("tracklets");
+      auto triggerrecords =
+          ctx.inputs().get<gsl::span<o2::trd::TriggerRecord>>("triggers");
       // std::cout << "Tracklets per time frame: " << tracklets.size();
       mTrackletsPerTimeFrame->Fill(tracklets.size());
       mTrackletsPerTimeFrameCycled->Fill(tracklets.size());
       mTriggersPerTimeFrame->Fill(triggerrecords.size());
-      for (auto& trigger : triggerrecords) {
+      if (!digits.size() == 0) {
+        mEventsPerTimeFrame->Fill(triggerrecords.size());
```

Comment:
You are only counting calibration triggers, what about calibration triggers (digits size >0) ?
If you want to only count digit events then it should be in the digits task, not the tracklets task.

Reply:
also conventionally you would do != instead of the if statement preceding this line.

Reply:
> You are only counting calibration triggers, what about pure physics triggers (digits size ==0 ) ? If you want to only count digit events then it should be in the digits task, not the tracklets task.

Hello, There is already one histogram in Tracklet task where we have events per time frame (not any condition on digits). But here we need to do events with digits per time frame. That is why I have included check on digits. But now I checked digit task and there I see that digit check is already there. So I only need to fill the histogram in digit task w/o digits check?
![image](https://user-images.githubusercontent.com/104899995/195050555-979efa93-73f9-4fcb-8669-09a3b35942b1.png)

Reply:
yes exactly. The lines after the continue have a non zero digit vector size.

Reply:
Ok thanks for your reply